x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -554,6 +565,26 @@ public AnalysisConfig build() {
                 modelPruneWindow);
         }
 
+        private void verifyModelPruneWindow() {
+            if (modelPruneWindow == null) {
+                return;
+            }
+
+            long modelPruneWindowSecs = modelPruneWindow.seconds();
+            long bucketSpanSecs = bucketSpan.seconds();
+
+            if (modelPruneWindowSecs % bucketSpanSecs != 0) {
+                throw ExceptionsHelper.badRequestException(MODEL_PRUNE_WINDOW.getPreferredName() + " [" + modelPruneWindow.toString() + "]"
+                    + " must be a multiple of " + BUCKET_SPAN.getPreferredName() + " [" + bucketSpan.toString() + "]");
+            }
```

Comment:
If this is a thing we are enforcing, it makes sense to me to make this `model_prune_window_buckets` and take a whole, positive value indicating the number of buckets in the past to look at. 

Forcing somebody to supply a timevalue when we require that value to always be a bucket multiple doesn't make sense to me

Reply:
@droberts195 ^ ? What say you?

Reply:
We decided in a team meeting about a month ago that we'd just have one setting that was an amount of time rather than a number of buckets.

I think it's quite unlikely this error will be thrown in practice when you consider sensible values for pruning are of the order of days, not minutes, and most bucket spans divide exactly into a day.

So I think it's OK as-is.

I definitely don't think we should change this retention period to be a number of buckets.  All the other retention periods are measured in time, not buckets.

What we could do is just enforce >= 2 * bucket span and not an exact multiple.  The C++ now rounds up so everything will work out OK.

But, like I said, I suspect in practice the detail here won't make any difference to anybody, because real users always choose nice round numbers even when they're not forced to.